homeassistant/components/risco/alarm_control_panel.py

```diff
@@ -165,7 +179,62 @@ async def _arm(self, mode, code):
         else:
             await self._call_alarm_method(risco_state)
 
+    async def _call_alarm_method(self, method, *args):
+        raise NotImplementedError
+
+
+class RiscoCloudAlarm(RiscoAlarm, RiscoEntity):
+    """Representation of a Risco partition."""
+
+    def __init__(
+        self, coordinator, partition_id, code: str, options: dict[str, Any]
+    ) -> None:
+        """Init the partition."""
+        super().__init__(
+            partition_id=partition_id,
+            partition=coordinator.data.partitions[partition_id],
+            coordinator=coordinator,
+            code=code,
+            options=options,
+        )
+        self._attr_unique_id = f"{self._risco.site_uuid}_{partition_id}"
+        self._attr_name = f"Risco {self._risco.site_name} Partition {partition_id}"
```

Comment:
If this represents the main device feature of this device we should set the entity name to `None` while setting `_attr_has_entity_name` to `True`.

https://developers.home-assistant.io/docs/core/entity#entity-naming